src/main/java/book/store/repository/shoppingcart/ShoppingCartRepository.java

```diff
@@ -0,0 +1,14 @@
+package book.store.repository.shoppingcart;
+
+import book.store.model.ShoppingCart;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @Query("select sc from ShoppingCart sc "
+            + "left join fetch sc.cartItems "
+            + "join fetch sc.user u ")
+    ShoppingCart findCartWithItems(Long userId);
```

Comment:
Not sure about the name for this query - because you're querying by a param different than identifier, IMO "findCardWithItemsByUserId" would be more explicit